ext/os/lib.rs

```diff
@@ -78,55 +78,13 @@ deno_core::extension!(
   ],
   esm = ["30_os.js", "40_signals.js"],
   options = {
-    exit_code: ExitCode,
+    exit_code: Option<ExitCode>,
   },
   state = |state, options| {
-    state.put::<ExitCode>(options.exit_code);
-    #[cfg(unix)]
-    {
-      state.put(ops::signal::SignalState::default());
+    if let Some(exit_code) = options.exit_code {
+      state.put::<ExitCode>(exit_code);
```

Comment:
This change was proposed in https://github.com/denoland/deno/pull/28693#pullrequestreview-2775083729 and I'm not a fan of it. It turns a compiler enforced invariant into something you need to know. -1 from me